lib/requestOptions.js

```diff
@@ -24,29 +24,34 @@ function parseHost(Container) {
   var host = Container.params.host;
   var req =  Container.user.req;
   var options = Container.options;
-  host = (typeof host === 'function') ? host(req) : host.toString();
-
-  if (!host) {
-    return new Error('Empty host parameter');
-  }
-
-  if (!/http(s)?:\/\//.test(host)) {
-    host = 'http://' + host;
-  }
-
-  var parsed = url.parse(host);
-
-  if (!parsed.hostname) {
-    return new Error('Unable to parse hostname, possibly missing protocol://?');
-  }
-
-  var ishttps = options.https || parsed.protocol === 'https:';
-
-  return {
-    host: parsed.hostname,
-    port: parsed.port || (ishttps ? 443 : 80),
-    module: ishttps ? https : http,
-  };
+  host = (typeof host === 'function') ? host(req) : host;
```

Comment:
dropped toString() here.  investigate why.

Reply:
I don't think the `toString()` is needed.  Anyways, I moved it to line [35](https://github.com/villadora/express-http-proxy/pull/285/files#diff-66ad66b7914d1efd19723774d3d91cd8R35) to support backward compatibility.